src/edu/csus/ecs/pc2/ui/team/CreateTeamsTSV.java

```diff
@@ -0,0 +1,268 @@
+     */
+    private String outputFileName = TEAMS_TSV_FILENAME;
+
+    /**
+     * Count for group ids.
+     */
+    private Map<String,Integer> groupMap = new HashMap<String, Integer>();
+
+    private String inputFileName = TEAMS_JSON_FILENAME;
+
+    public CreateTeamsTSV(String[] args) throws CommandLineErrorException {
+        loadVariables(args);
+    }
+
+    @SuppressWarnings("unused")
+    private void dprint(String string) {
+        if (debugMode) {
+            System.out.println(string);
+        }
+    }
+
+    protected void loadVariables(String[] args) throws CommandLineErrorException {
+
+        String[] argsWithValues = { OUTPUT_FILE_KEY, INPUT_FILE_KEY};
+        ParseArguments arguments = new ParseArguments(args, argsWithValues);
+
+        if (arguments.isOptPresent("--help")) {
+            usage();
+            System.exit(4);
+        }
+
+        debugMode = arguments.isOptPresent("--debug");
+        
+        
+        allowOverWrite = arguments.isOptPresent(OVERWRITE_FILE_KEY);
+        
+
+        if (debugMode) {
+            arguments.dumpArgs(System.err);
+        }
+
+        if (arguments.isOptPresent(OUTPUT_FILE_KEY)) {
+            outputFileName = arguments.getOptValue(OUTPUT_FILE_KEY);
+        }
+        
+        if (arguments.isOptPresent(INPUT_FILE_KEY)) {
+            inputFileName = arguments.getOptValue(INPUT_FILE_KEY);
+        }
+        
+
+    }
+
+    private void fatalError(String errorMessage) {
+        System.err.println("Fatal error " + errorMessage);
+        System.exit(FAILURE_EXIT_CODE);
+
+    }
+
+    private static void usage() {
+        String[] usage = { //
+                "", //
+                "Usage CreateTeamsTSV [--help] [-f] [--of outputfilename] [--if teams_file.json]", //
+                "", //
+                "Purpose: read teams.json file create teams.tsv and groups.tsv files.", //
+                "", //
+                "This will not overwrite output file, use -f to overwrite output file (and "+GROUPS_TSV_FILENAME+")", //
+                "", //
+                "options:", //
+                "--of file  - alternate output filename, default output filename is " + TEAMS_TSV_FILENAME, //
+                "--if file  - alternate input JSON filename, default  filename is " + TEAMS_JSON_FILENAME, //
+                     "-f   overwrite output file, by default will not overwrite output file.", //
+                "", //
+                "On success exit code will be " + SUCCESS_EXIT_CODE, //
+                "Any other exit code is an error.", //
+                "", //
+        };
+
+        for (String s : usage) {
+            System.out.println(s);
+        }
+    }
+
+    protected void writeTSVFile() throws Exception {
+
+        if (!allowOverWrite  && new File(outputFileName).exists()) {
+            fatalError("Output file exists (" + outputFileName + "), will not overwrite.  Use " + OUTPUT_FILE_KEY + " to write to a different file");
```

Comment:
It seems like this message is incomplete.  It mentions that OUTPUT_FILE_KEY can be used to write to a _different_ file -- but another common reason for getting to this point would be failure to use OVERWRITE_FILE_KEY to overwrite the _existing_ file.  It seems like that should be mentioned in the error message as well...